compiler/optimizer/CopyPropagation.cpp

```diff
@@ -728,6 +728,14 @@ int32_t TR_CopyPropagation::perform()
             replaceCopySymbolReferenceByOriginalIn(copySymbolReference, equivalentDefNode, useNode, defNode);
             usesToBeFixed[useNode->getUseDefIndex()] = equivalentDefs[defIndex];
 
+            if (copySymbolReference->hasKnownObjectIndex() &&
+                !originalSymbolReference->hasKnownObjectIndex() &&
+                !useNode->hasKnownObjectIndex())
+               {
+               useNode->setKnownObjectIndex(copySymbolReference->getKnownObjectIndex());
+               dumpOptDetails(comp(), "%s   Set known-object obj%d on Use node : %p\n", OPT_DETAILS, copySymbolReference->getKnownObjectIndex(), useNode);
+               }
```

Comment:
Is it enough to catch it here, or should this also be done after the call to `replaceCopySymbolReferenceByOriginalIn()` in the "Copy2 Propagation" path below?

Reply:
Yes, we should do it. Maybe I should move my code inside `replaceCopySymbolReferenceByOriginalIn() `? @vijaysun-omr what do you think?

Reply:
That makes sense, all calls to `setSymbolReference` are in that routine.